vm/src/builtins/map.rs

```diff
@@ -24,13 +24,19 @@ impl PyValue for PyMap {
 
 #[pyimpl(with(PyIter), flags(BASETYPE))]
 impl PyMap {
+    #[pymethod(magic)]
+    fn length_hint(&self, vm: &VirtualMachine) -> PyResult<usize> {
+        self.iterators.iter().try_fold(0, |prev, cur| {
+            let cur = iterator::length_hint(vm, cur.clone())?.unwrap_or(0);
+            let max = std::cmp::max(prev, cur);
+            Ok(max)
+        })
+    }
+
     #[pyslot]
-    fn tp_new(
-        cls: PyTypeRef,
-        function: PyObjectRef,
-        iterables: Args,
-        vm: &VirtualMachine,
-    ) -> PyResult<PyRef<Self>> {
+    fn tp_new(cls: PyTypeRef, mut args: FuncArgs, vm: &VirtualMachine) -> PyResult {
```

Comment:
I think turning `FnucArgs`-less `tp_new`  to use `FuncArgs` is generally not a good idea. `SlotConstructor` can keep the original argument without code change by defining proper `SlotCosntructor::Args`, mostly just tuple of the types.

Reply:
I strongly agree with it. to be honest i didn't know the way using handles.